src/Interpreters/executeQuery.cpp

```diff
@@ -783,6 +783,7 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
             catch (const Exception & e)
             {
                 if (e.code() == ErrorCodes::SYNTAX_ERROR)
+                    /// Don't print the original query text because it may contain sensitive data.
```

Comment:
Is it really the case, after `formatWithPossiblyHidingSensitiveData` above?

Reply:
By "original query" I meant `ast`, which didn't go through `formatWithPossiblyHidingSensitiveData`. So we print `formatted1`, which did. This comment was my attempt at guessing the intent of the original author of this code. I also considered that it's !NDEBUG-only, but was still too afraid to change it. Yeah, printing `ast` would be useful, but so far I've been able to guess the problem from just `formatted1` ~2/2 times.

Reply:
What do you think about adding a knob for it then? https://github.com/ClickHouse/ClickHouse/pull/67776